PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP is not portable

While musl libc supports recursive mutexes, the initaliziation macro appears to be specific to glibc, causing a build error:
```
/home/lambda/olinuxino/openwrt/build_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/elektra-0.8.15dev/src/plugins/resolver/resolver.c:52:49: error: 'PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP' undeclared here (not in a function)
```
See this link for the same issue on an other project: https://github.com/Itseez/opencv/issues/5476

markus2330 writes:
> @haraldg You can easily avoid the issue by removing all resolver variants that contain `m` (for mutex).

I tried it with:
        -DKDB_DEFAULT_RESOLVER=resolver_l_h_b \
        -DPLUGINS="DEFAULT;-resolver;resolver_l_h_b"

but got an error:
CMake Error at cmake/ElektraCache.cmake:338 (message):
  selected default resolver (resolver_l_h_b) is not selected in PLUGINS,
  please change KDB_DEFAULT_RESOLVER or PLUGINS

--- a/src/plugins/resolver/resolver.c
+++ b/src/plugins/resolver/resolver.c
@@ -46,10 +46,12 @@
 # define statNanoSeconds(status) status.st_mtim.tv_nsec
 #endif
 
-#if defined(__APPLE__) && defined(__MACH__)
+#ifdef ELEKTRA_LOCK_MUTEX
+#ifndef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
 static pthread_mutex_t elektra_resolver_mutex = PTHREAD_RECURSIVE_MUTEX_INITIALIZER;
 #else
 static pthread_mutex_t elektra_resolver_mutex = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+#endif
 #endif
 
 static void resolverInit (resolverHandle *p, const char *path)

